content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+
+In parsnip, the functions `frequency_weights()` and `importance_weights()` can be used to set the weights: 
+
+```{r vec-wts}
+# For the UC admissions data
+ucb$n <- frequency_weights(ucb$n)
+ucb
+
+# For a general vector of importance weights: 
+importance_weights(round(runif(10), 2))
+```
+
+The class of these objects tell packages like recipes and yardstick if their values should be used for preprocessing and performance metrics, respectively: 
+
+* Importance weights only affect the model fit and _supervised_ recipes steps. They are not used with yardstick functions.
+
+* Frequency weights are used for all parts of the preprocessing, model fitting, and performance estimation operations. 
+
+Currently, these are the only classes implemented. We are doing a lot of reading on how the analysis of survey data should use case weights and how we can enable this and other data analysis use cases. [We'd love to hear from you](https://community.rstudio.com/) if you have expertise in this area. 
+
+## About resampling
+
+This is a topic that we are still unsure about. We definitively think that importance weights should not affect how the data are split or resampled. 
+
+Frequency weights are more complex. Suppose we are using 10-fold cross-validation with the logistic regression on the UCB admission data, should we:
+
+* Have all the case weights be placed into either the modeling or holdout set? 
+* Fractionally, split the case weights into both the modeling and holdout data? 
+
+For the latter case, suppose a row of data has a case weight of 100 and we use 10-fold cross-validation. We would always put 90 of those 100 into the modeling data set and the other 10 into the holdout. This seems to be consistent with the sampling of the data and is what would happen if there were actually 100 rows in the data (instead of one row with a case weight of 100). However, it does raise questions regarding data leakage by just re-predicting the same data that went into the model. 
+
+This is also an area where we'd like [community feedback](https://community.rstudio.com/). 
+
+## Tidymodels syntax
+
+Let's work through an example. We'll use some data simulated with a severe class imbalance. These functions are in the new version of the modeldata package.
+
+```{r sim-data}
+set.seed(1)
+training_sim <- sim_classification(5000, intercept = -25) 
+training_sim %>% count(class)
+```
+
+If we would like to encourage models to more accurately predict the minority class, we can give these samples a much larger weight in the analysis
+
+```{r sim-wts}
+training_sim <-
+  training_sim %>% 
+  mutate(
+    case_wts = ifelse(class == "class_1", 60, 1),
+    case_wts = importance_weights(case_wts)
+  )
+```
+
+We strongly advise that users set the case weight column before any other tidymodels functions are used. This ensures that they are handled correctly in the analyses that follow. 
```

Comment:
If there's a simple cautionary tale, would be nice to add to the end of this. Something like "Not doing so could result in ..."